float64[7] q
float64[7] dq
